src/nspawn/nspawn.c

```diff
@@ -2218,6 +2285,29 @@ static int setup_machine_id(const char *directory) {
         return 0;
 }
 
+static int recursive_chown(const char *directory, uid_t shift, uid_t range) {
+        int r;
+
+        assert(directory);
+
+        if (arg_userns_mode == USER_NAMESPACE_NO || !arg_userns_chown)
+                return 0;
+
+        r = path_patch_uid(directory, arg_uid_shift, arg_uid_range);
+        if (r == -EOPNOTSUPP)
+                return log_error_errno(r, "Automatic UID/GID adjusting is only supported for UID/GID ranges starting at multiples of 2^16 with a range of 2^16.");
+        if (r == -EBADE)
+                return log_error_errno(r, "Upper 16 bits of root directory UID and GID do not match.");
+        if (r < 0)
+                return log_error_errno(r, "Failed to adjust UID/GID shift of OS tree: %m");
```

Comment:
I get:
```
Failed to adjust UID/GID shift of OS tree: Read-only file system
```

It would be nice to tell me which filesystem / directory it is ;)

Strace tells me:

```
[pid 20954] openat(6, "sys", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_NOFOLLOW|O_NOATIME|O_CLOEXEC) = 10
[pid 20954] +++ exited with 1 +++
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=20954, si_uid=0, si_status=1, si_utime=0, si_stime=6} ---
unlink("/run/systemd/nspawn-uid/1536360448") = 0
syscall_326(0x6, 0, 0x1, 0, 0x7fffffffffffffff, 0) = -1 (errno 38)
Selected user namespace base 1536360448 and range 65536.
Failed to adjust UID/GID shift of OS tree: Read-only file system
open("/run/systemd/nspawn/propagate/rkt-f90fcda1-103c-4d64-8956-67dd501eb5af", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_NOFOLLOW|O_NOATIME|O_CLOEXEC) = -1 ENOENT (No such file or directory)
unlink("/var/lib/rkt/pods/run/f90fcda1-103c-4d64-8956-67dd501eb5af/stage1/.#rootfs.lck") = 0
unlink("/run/systemd/nspawn/locks/inode-64769:893055") = 0
+++ exited with 1 +++
```